### PR TITLE
Feat/task 4714

### PR DESCRIPTION
**Pull request by sureshkrishnan-mr**
[XXXX-<Title> - Please use the Work Item number and Title as PR Name, not subtasks]

#### 📲 What

The proliferation of Swagger annotations means that there is a lot of duplication, and it makes it difficult to properly compose the controller operations into a single class.

#### 🤔 Why

Use Java annotation(s) to remove duplicated Swagger annotations from controller methods.
#### 🛠 How

More in-depth discussion of the change or implementation.

#### 👀 Evidence

Screenshots / external resources / links / etc.
Link to documentation updated with changes impacted in the PR

#### 🕵️ How to test

Notes for QA

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v1/DeleteItemController.java [hunk 1]
```diff
@@ -1,68 +1,41 @@
...
-        @ApiResponse(
-            responseCode = "404",
-            description = "Resource not found",
-            content =
-                @Content(
-                    mediaType = "application/json",
-                    schema = @Schema(implementation = ErrorResponse.class)))
-      })
+      operationId = "DeleteMenuItem")
+  @ApiResponse(
```
Comment (sdavis-amido) [inline java/src/main/java/com/amido/stacks/workloads/menu/api/v1/DeleteItemController.java:31]: Any reason why this 404 could not be put on the base annotation (@DeleteAPIResponses)?

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v1/UpdateCategoryController.java [hunk 2]
```diff
@@ -1,74 +1,46 @@
...
-        @ApiResponse(
-            responseCode = "409",
-            description = "Conflict, an item already exists",
-            content =
-                @Content(
-                    mediaType = "application/json",
-                    schema = @Schema(implementation = ErrorResponse.class)))
-      })
+      operationId = "UpdateMenuCategory")
+  @ApiResponse(
```
Comment (sdavis-amido) [inline java/src/main/java/com/amido/stacks/workloads/menu/api/v1/UpdateCategoryController.java:36]: Same question, why can this not be added to the base annotation?

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v1/DeleteItemController.java (hunk 1 above)
Comment (sdavis-amido) [inline java/src/main/java/com/amido/stacks/workloads/menu/api/v1/DeleteItemController.java:31]: Any reason why this 404 could not be put on the base annotation (@DeleteAPIResponses)?

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v1/UpdateItemController.java [hunk 3]
```diff
@@ -1,73 +1,45 @@
...
-        @ApiResponse(
-            responseCode = "404",
-            description = "Resource not found",
-            content =
-                @Content(
-                    mediaType = "application/json",
-                    schema = @Schema(implementation = ErrorResponse.class)))
-      })
+      operationId = "UpdateMenuItem")
+  @ApiResponse(
```
Comment (sdavis-amido) [inline java/src/main/java/com/amido/stacks/workloads/menu/api/v1/UpdateItemController.java:35]: Same for this, why not base?

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v1/UpdateMenuController.java [hunk 4]
```diff
@@ -1,72 +1,45 @@
...
-        @ApiResponse(
-            responseCode = "409",
-            description = "Conflict, an item already exists",
-            content =
-                @Content(
-                    mediaType = "application/json",
-                    schema = @Schema(implementation = ErrorResponse.class)))
-      })
+      description = "Update a menu with new information")
+  @ApiResponse(
```
Comment (sdavis-amido) [inline java/src/main/java/com/amido/stacks/workloads/menu/api/v1/UpdateMenuController.java:35]: Same again, why not base?

**Review by sdavis-amido — CHANGES_REQUESTED** [review]

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v1/DeleteMenuController.java [hunk 5]
```diff
@@ -1,72 +1,41 @@
...
-        @ApiResponse(
-            responseCode = "404",
-            description = "Resource not found",
-            content =
-                @Content(
-                    mediaType = "application/json",
-                    schema = @Schema(implementation = ErrorResponse.class)))
-      })
+      operationId = "DeleteMenu")
+  @ApiResponse(
```
[the remark this answers is not in the document]
Comment (sureshkrishnan-mr) [reply java/src/main/java/com/amido/stacks/workloads/menu/api/v1/DeleteMenuController.java:31]: On the main branch, deleteItem and deleteMenu both have 404 status but deleteCategory does not have 404 (not sure why).
I think we can add 404 for deleteCategory and move this to base annotation?

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v1/DeleteItemController.java (hunk 1 above)
Comment (sdavis-amido) [reply java/src/main/java/com/amido/stacks/workloads/menu/api/v1/DeleteItemController.java:31]: Yes - I don't see why Delete would not have a 404 as well, it's just as likely to occur there. Would agree that you add it, as you suggest.

Comment (ElvenSpellmaker) [reply java/src/main/java/com/amido/stacks/workloads/menu/api/v1/DeleteItemController.java:31]: We need to find out what the intention was, the API has to be parity with dotnet too, so it's possible it was copied over from that. (Although maybe dotnet is wrong if so)

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v1/UpdateMenuController.java (hunk 4 above)
Comment (sureshkrishnan-mr) [reply java/src/main/java/com/amido/stacks/workloads/menu/api/v1/UpdateMenuController.java:35]: UpdateItem does not seem to have 409 status on the main branch (again not sure why). I believe we can add 409 and move this to base annotation?

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v1/UpdateItemController.java (hunk 3 above)
Comment (sureshkrishnan-mr) [reply java/src/main/java/com/amido/stacks/workloads/menu/api/v1/UpdateItemController.java:35]: Only updateItem seem to have 404 status on the main branch. updateMenu and updateMenuCategory does not have 404. I think we can add 404 for all of them and move his to base annotation?

**Review by sdavis-amido — APPROVED** [review]

**Review by sdavis-amido — APPROVED** [review]

**Pull request merged**